src/Raven.Server/Documents/Handlers/Debugging/ProcStatsHandler.cs

```diff
@@ -0,0 +1,210 @@
+            {
+                var djv = CpuStatsInternal();
+
+                using (var write = new BlittableJsonTextWriter(context, ResponseBodyStream()))
+                {
+                    context.Write(write, djv);
+                }
+                return Task.CompletedTask;
+            }
+        }
+
+        [RavenAction("/admin/debug/proc/stats", "GET", AuthorizationStatus.Operator, IsDebugInformationEndpoint = true)]
+        public Task ProcStats()
+        {
+            using (ServerStore.ContextPool.AllocateOperationContext(out JsonOperationContext context))
+            {
+                var djv = ProcStatsInternal();
+
+                using (var write = new BlittableJsonTextWriter(context, ResponseBodyStream()))
+                {
+                    context.Write(write, djv);
+                }
+                return Task.CompletedTask;
+            }
+        }
+
+        private DynamicJsonValue CpuStatsInternal()
+        {
+            var proc = Process.GetCurrentProcess();
+
+            var djaCpu = new DynamicJsonArray();
+            var djvCpu = new DynamicJsonValue();
+
+            djvCpu["ProcessName"] = proc.ProcessName;
+            djvCpu["ProcessorAffinity"] = proc.ProcessorAffinity.ToInt64();
+            djvCpu["PrivilegedProcessorTime"] = proc.PrivilegedProcessorTime;
+            djvCpu["TotalProcessorTime"] = proc.TotalProcessorTime;
+            djvCpu["UserProcessorTime"] = proc.UserProcessorTime;
+
+            djaCpu.Add(djvCpu);
+
+            var djaThreadpool = new DynamicJsonArray();
+            var djvThreadpool = new DynamicJsonValue();
+
+            ThreadPool.GetAvailableThreads(out var workerThreads, out var completionPortThreads);
+            djvThreadpool["AvailableThreadPoolWorkerThreads"] = workerThreads;
+            djvThreadpool["AvailableThreadPoolCompletionPortThreads"] = completionPortThreads;
+            ThreadPool.GetMinThreads(out workerThreads, out completionPortThreads);
+            djvThreadpool["MinThreadPoolWorkerThreads"] = workerThreads;
+            djvThreadpool["MinThreadPoolCompletionPortThreads"] = completionPortThreads;
+            ThreadPool.GetMaxThreads(out workerThreads, out completionPortThreads);
+            djvThreadpool["MaxThreadPoolWorkerThreads"] = workerThreads;
+            djvThreadpool["MaxThreadPoolCompletionPortThreads"] = completionPortThreads;
+
+            djaThreadpool.Add(djvThreadpool);
+
+            return new DynamicJsonValue
+            {
+                ["CpuStats"] = djaCpu,
+                ["ThreadPoolStats"] = djaThreadpool
+            };
+        }
+
+        private DynamicJsonValue ProcStatsInternal()
+        {
+            var proc = Process.GetCurrentProcess();
+
+            var dja = new DynamicJsonArray();
+            var djv = new DynamicJsonValue();
+            foreach (var prop in typeof(Process).GetProperties())
```

Comment:
Why are you doing this dynamically?

Reply:
The /debug/cpu/stats is a summarized (not dynamically generated) end point.  I added this /debug/proc/stats dynamically as it contains a lot of info which can be useful when debugging, and is also changing from one to another dotnet release.   I though reflection suits here.  Should I manually write each property?

Reply:
Yes, better that way. Feel free to generate the code, but it is much better to read it in this manner.

Reply:
done